### PR TITLE
fix: prevent stack overflow on validation resolvers generation

### PR DESCRIPTION
**Pull request by GaelS**
In order to fix https://github.com/profusion/apollo-validation-directives/issues/41 

Hope it fits your code requirements.

### CODE REVIEW

File: lib/ValidateDirectiveVisitor.ts [hunk 1]
```diff
@@ -198,9 +197,14 @@ const checkMustValidateInput = (
   }
 
   if (finalType instanceof GraphQLInputObjectType) {
-    return Object.values(finalType.getFields()).some(
-      checkMustValidateInputField,
-    );
+    const fieldsToCheck = Object.values(finalType.getFields()).filter(field => {
+      if (field.type instanceof GraphQLList) {
```
Comment (cabelitos) [inline lib/ValidateDirectiveVisitor.ts:201]: Instead of doing this filter, can't we check this inside the checkMustValidateInput function?

**Comment by cabelitos** [comment]
Thank you very much for your contribution

File: lib/ValidateDirectiveVisitor.ts (hunk 1 above)
Comment (GaelS) [reply lib/ValidateDirectiveVisitor.ts:201]: Something like that :thinking:  ?

```
const checkMustValidateInputField = (
  field: ValidatedGraphQLInputField,
): boolean => {
  // istanbul ignore next (shouldn't reach as addContainerEntryValidation would mark parent, but be safe)
  if (field.validation) {
    return true;
  }
  if (field.type instanceof GraphQLInputObjectType) {
    return !Object.values(field.type.getFields()).some(
      nestedField => nestedField.type === field.type,
    );
  }
  if (field.type instanceof GraphQLList) {
    return field.type.ofType !== field.type;
  }
  // eslint-disable-next-line @typescript-eslint/no-use-before-define
  return checkMustValidateInput(field.type);
};
```

Comment (GaelS) [reply lib/ValidateDirectiveVisitor.ts]: I just pushed something

**Comment by cabelitos** [comment]
@GaelS taking a second read on your commit, IMO it will be better to remove the 81c5667 commit and leave only the original one. Could you please do that so we can merge it? thank you.

Sorry for the double work,

**Comment by GaelS** [comment]
no problem :) it's done

**Review by cabelitos — APPROVED** [review]

**Comment by cabelitos** [comment]
@GaelS  ty very much, I plan to make a release later today

**Comment by cabelitos** [comment]
@GaelS published on NPM under version 2.1.2

Thank you very much for your help!

**Comment by GaelS** [comment]
Awesome, thank you !

**Pull request merged**